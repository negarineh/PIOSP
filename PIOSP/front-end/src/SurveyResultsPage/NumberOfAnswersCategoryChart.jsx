import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import '../components/stylesheets/menuStyle.css';
import 'react-tabs/style/react-tabs.css';
import "react-table/react-table.css";

import {Pie} from 'react-chartjs-2';

// Pie chart base on total answers per category
class NumberOfAnswersCategoryChart extends React.Component {

    constructor(props) {
        super(props);
        this.state = {
          category:[],
          chartData1: {},
        }; 
      }

    componentDidMount() {

       fetch('/equal')
       .then(data => data.json())
       .then((data =>{ 
        const chartData1 = {
            labels: data.map(k => k.category),
            datasets: [
              {
                label: 'Number of Answes',
                data: data.map(d => d.count),
                backgroundColor: [
                    '#FF6384',
                    '#36A2EB',
                    '#FFCE56',
                    '#FF4136',
                    '#3D9970',
                    '#85144b',
               ],
              },
            ]
          };
  
          this.setState({ chartData1 });
       }))
       .catch(err => console.log(err));

    }
  
    render() {
        const { chartData1} = this.state;

        return (
            <div >
                <Pie data={chartData1}/>
            </div>
        );
    }
}

function mapStateToProps(state) {
    const { users, authentication } = state;
    const { user } = authentication;
    return {
        user,
        users
    };
}

const connectedHomePage = connect(mapStateToProps)(NumberOfAnswersCategoryChart);
export { connectedHomePage as NumberOfAnswersCategoryChart };

NumberOfAnswersCategoryChart.defaultProps = {
    chartData1: {}
}

NumberOfAnswersCategoryChart.propTypes = {
    chartData1: PropTypes.object.isRequired,
}