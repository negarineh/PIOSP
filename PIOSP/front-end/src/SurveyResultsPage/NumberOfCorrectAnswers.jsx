import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import '../components/stylesheets/menuStyle.css';
import 'react-tabs/style/react-tabs.css';
import "react-table/react-table.css";

import {Bar} from 'react-chartjs-2';

//Bar chart base on correct and incorrect answers per category
class NumberOfCorrectAnswers extends React.Component {

    constructor(props) {
        super(props);
        this.state = {
          category:[],
          correct:[],
          chartData1: {},
        }; 
      }

    componentDidMount() {

       fetch('/totalCorrectAnswers')
       .then(data => data.json())
       .then((data =>{ 
        const chartData1 = {
            datasets: [
              {
                label: 'Correct Answers',
                data: data.map(d => d.correct),
                backgroundColor: [
                    'green',
               ],
              },
              {
                label: 'InCorrect Answers',
                data: data.map(d => d.inCorrect),
                backgroundColor: [
                    'red',
               ],
              }
            ]
          };
  
          this.setState({ chartData1 });
       }))
       .catch(err => console.log(err));

    }
  
    render() {
        const {chartData1} = this.state;

        return (
            <div >
                <Bar data={chartData1}/>
            </div>
        );
    }
}

function mapStateToProps(state) {
    const { users, authentication } = state;
    const { user } = authentication;
    return {
        user,
        users
    };
}

const connectedHomePage = connect(mapStateToProps)(NumberOfCorrectAnswers);
export { connectedHomePage as NumberOfCorrectAnswers };

NumberOfCorrectAnswers.defaultProps = {
    chartData1: {}
}

NumberOfCorrectAnswers.propTypes = {
    chartData1: PropTypes.object.isRequired,
}