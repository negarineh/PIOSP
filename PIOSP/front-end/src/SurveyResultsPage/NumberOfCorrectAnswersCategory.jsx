import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import '../components/stylesheets/menuStyle.css';
import 'react-tabs/style/react-tabs.css';
import "react-table/react-table.css";

import { Dropdown } from 'semantic-ui-react';
import { BarChart, Bar, XAxis, YAxis, CartesianGrid, Tooltip, Legend, LineChart, Line, ResponsiveContainer} from 'recharts';

import {Doughnut, Pie} from 'react-chartjs-2';

// we have a dropdown list to show number of correct and incorrect answers base on category
// to visualize we have options between doughnut chart, bar chart , line chart ,and pie chart 
const tagOptionsMenu = [
    {
      text: 'Doughnut Chart ',
      value: 'Doughnut Chart',
      icon: {name:'table', color:'black'}
    },
    {
      text: 'Pie Chart',
      value: 'Pie Chart',
      icon: {name:'table', color:'black'},
    },
    {
      text: 'Bar Chart',
      value: 'Bar Chart',
      icon: {name:'table', color:'black'},
    },
    {
      text: 'Line Chart',
      value: 'Line Chart',
      icon: {name:'table', color:'black'},
    },
  ];

  const option1 = {
    title: {
      display: true,
      text: 'Outer chart: Total number of answer per category, Inner Chart: Number of correct answer per categoey'
  },
  legend:{
    display: true,
      text: 'ETL Load Chart'
  }
  };

class NumberOfCorrectAnswersCategory extends React.Component {

    constructor(props) {
        super(props);
        this.state = {
          expanded: false,
          photo:'',
          answers:[],
          category:[],
          correct:[],
          chartData1: {},
          chartData2: {},
          activeItem: 'home',
          visible: false,
          option:'',
        }; 
      }

    componentDidMount() {
        
      fetch('/notEqual')
        .then(data => data.json())
        .then((data) => { this.setState({ correct: data});
      }).catch(err => console.log(err));

       fetch('/equal')
       .then(data => data.json())
       .then((data =>{ 
        const chartData1 = {
            labels: data.map(k => k.category),
            datasets: [
              {
                label: 'Number of Answerrs',
                data: data.map(d => d.count),
                backgroundColor: [
                    '#FF6384',
                    '#36A2EB',
                    '#FFCE56',
                    '#FF4136',
                    '#3D9970',
                    '#85144b',
               ],
              },
              {
                label: 'Correct Answers',
                data: data.map(d => d.correct),
                backgroundColor: [
                    '#FF6384',
                    '#36A2EB',
                    '#FFCE56',
                    '#FF4136',
                    '#3D9970',
                    '#85144b',
               ],
              }
            ]
          }
          const chartData2 = {
            labels: data.map(k => k.category),
            datasets: [
              {
                label: 'Number of Correct Answers Base on Category',
                data: data.map(d => d.correct),
                backgroundColor: [
                    '#FF6384',
                    '#36A2EB',
                    '#FFCE56',
                    '#FF4136',
                    '#3D9970',
                    '#85144b',
               ],
              }
            ]
          };
  
          this.setState({ chartData1 , chartData2 });
       }))
       .catch(err => console.log(err));

    }

    toggleVisibility = () => this.setState({ visible: !this.state.visible })
  
    handleItemClick = (e, { name }) => {
      this.setState({ activeItem: name });
    }
  
  
    handleChange = (e, { name, value }) => 
    {
      this.setState({ [name]: value , option: value});
      if (value === 'Profile')
          alert("Profile");
      if (value === 'SignOut')
          alert('SignOut');
    }
  
    render() {
        const { correct, chartData1, chartData2, option} = this.state;

        return (
            <div >
              <Dropdown
                selection
                name='default'
                options={tagOptionsMenu}
                placeholder='Choose the Graph'
                onChange={this.handleChange.bind(this)}
                onClick={this.handleItemClick.bind(this)}
                fluid
              />
              {option ==='Doughnut Chart'?
                <Doughnut data={chartData1} options={option1}/>:
                <span></span> 
                }
                
                {option ==='Pie Chart'?
                <Pie data={chartData2}/>:
                <span></span> 
                }

                {option ==='Bar Chart'?
                <ResponsiveContainer width='100%' aspect={5.5/2.5}>
                    <BarChart width={600} height={300} data={correct}
                            margin={{top: 5, right: 30, left: 20, bottom: 5, color:"brown"}}>
                            <XAxis dataKey="category"/>
                            <YAxis/>
                            <CartesianGrid strokeDasharray="3 3"/>
                            <Tooltip/>
                            <Legend />
                            <Bar dataKey="correct"  fill="#8884d8" />
                            <Bar dataKey="inCorrect"  fill="#82ca9d" />
                    </BarChart>
                    </ResponsiveContainer>:
                <span></span> 
                }

                {option ==='Line Chart'?
                  <ResponsiveContainer width='100%' aspect={5.5/2.5}>
                    <LineChart width={600} height={600} data={correct} margin={{ top: 5, right: 30, left: 20, bottom: 5 }}>
                        <XAxis dataKey='category'/>
                        <YAxis/>
                        <CartesianGrid stroke="#eee" strokeDasharray="5 5"/>
                        <Tooltip/>
                        <Legend />
                        <Line type="monotone" datakey="count" stroke="#8884d8" activeDot={{r: 8}}/>
                        <Line type="monotone" dataKey="correct" stroke="#82ca9d"/>
                        <Line type="monotone" dataKey="Wasp" stroke="darkcyan" />
                        <Line type="monotone" dataKey="Fly" stroke="coral" />
                        <Line type="monotone" dataKey="Moth" stroke="crimson" />
                        <Line type="monotone" dataKey="DKnow" stroke="darkred" />
                    </LineChart>
                  </ResponsiveContainer>:
                <span></span> 
                }
            </div>
        );
    }
}

function mapStateToProps(state) {
    const { users, authentication } = state;
    const { user } = authentication;
    return {
        user,
        users
    };
}

const connectedHomePage = connect(mapStateToProps)(NumberOfCorrectAnswersCategory);
export { connectedHomePage as NumberOfCorrectAnswersCategory };

NumberOfCorrectAnswersCategory.defaultProps = {
    chartData1: {},
    chartData2: {},
    handleItemClick: () => {},
    handleChange: () => {}
}

NumberOfCorrectAnswersCategory.propTypes = {
    chartData1: PropTypes.object.isRequired,
    chartData2: PropTypes.object.isRequired,
    handleItemClick: PropTypes.func.isRequired,
    handleChange: PropTypes.func.isRequired,
}