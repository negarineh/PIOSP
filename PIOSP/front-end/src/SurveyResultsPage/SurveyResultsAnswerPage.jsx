import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { resultActions } from '../_actions';

import '../components/stylesheets/menuStyle.css';
import 'react-tabs/style/react-tabs.css';
import '../components/stylesheets/MainPage.css';

// Import React Table
import ReactTable from "react-table";
import "react-table/react-table.css";

import matchSorter from 'match-sorter';

import { Confirm, Button, Segment} from 'semantic-ui-react';
import {CSVLink} from 'react-csv';

// all answers for submit page in a table by query options like name, id, date,.etc
class SurveyResultsAnswerPage extends React.Component {

    constructor() {
        super();
        this.state = {
        //   datas: resultActions.getAllSurveyExperienceResults(),
          open: false,
          result: '',
          id1:'',
          answers:[],
        };
        this.show = this.show.bind(this);
        this.handleConfirm = this.handleConfirm.bind(this);
        this.handleCancel = this.handleCancel.bind(this);
      }
    
    componentDidMount() {

        fetch('/reportExperienceAnswers')
        .then(data => data.json())
        .then((data) => { this.setState({ answers: data });
      }).catch(err => console.log(err));
         
        this.props.dispatch(resultActions.getAllSurveyExperienceResults());
    }

    changeID(){
        var num =0;
        var newResults = this.state.answers;
        for (var i=0; i<= this.state.answers.length; i++)
            if(this.state.answers._id[i] === this.state.answers._id[i+1] )
                newResults._id[i] = num;
            else
            num = num+1;
        console.log(newResults)   ;      
    }

    show = (id) => this.setState({ open: true, id1: id });
    handleConfirm = () => this.setState({ result: 'confirmed', open: false });
    handleCancel = () => this.setState({ result: 'cancelled', open: false });

  
    render() {
        const { results  } = this.props;
        const { answers, open } = this.state;

        return (
            <div>
                <Segment style={{height: '638.5px'}}>
                    <h3>All answers:</h3>
                    {results.items &&
                        <div className="wrapper" >
                            <ReactTable style={{backgroundColor:"beige", maxHeight:'450px'}}
                                data={results.items}
                                    filterable
                                    defaultFilterMethod={(filter, row) =>
                                    String(row[filter.id]) === filter.value}
                            
                                columns={[
                                    {
                                        columns: [
                                            {
                                            Header: "ID",
                                            id: "_id",
                                            accessor: d => d._id,
                                                filterMethod: (filter, rows) =>
                                                matchSorter(rows, filter.value, { keys: ["_id"] }),
                                                filterAll: true
                                            },
                                            {
                                            Header: "Experience Rate",
                                            id: "experience",
                                            accessor: d => d.experience,
                                                filterMethod: (filter, rows) =>
                                                matchSorter(rows, filter.value, { keys: ["experience"] }),
                                                filterAll: true
                                            }
                                                ]
                                    }
                                    ,
                                    {
                                        columns: [
                                                {
                                                Header: "Participation",
                                                id: "participation",
                                                accessor: d => d.participation,
                                                filterMethod: (filter, rows) =>
                                                matchSorter(rows, filter.value, { keys: ["participation"] }),
                                                filterAll: true,
                                                },
                                                {
                                                Header: "Description",
                                                accessor: "description",
                                                id: "description",
                                                filterMethod: (filter, rows) =>
                                                matchSorter(rows, filter.value, { keys: ["description"] }),
                                                filterAll: true,
                                                },
                                        ]
                                    },
                                    {
                                    Header: "Created Date",
                                    id: "created_date",
                                    accessor: d => d.created_date.toString().slice(0,10),
                                    filterMethod: (filter, rows) =>
                                    matchSorter(rows, filter.value, { keys: ["created_date"] }),
                                    filterAll: true
                                    },
                                ]}
                                defaultPageSize={10}
                                className="-striped -highlight"
                            />
                        <Confirm 
                        open={open}
                        onCancel={this.handleCancel}
                        onConfirm={this.handleConfirm}/>

                    </div>        
                }
                    <Button style={{margin:'20px'}}>
                        <CSVLink data={answers} >Download Survey Results CSV file</CSVLink>
                    </Button>
                </Segment>
            </div>
        );
    }
}

function mapStateToProps(state) {
    const { answers, authentication, results } = state;
    const { user } = authentication;
    return {
        user,
        answers,
        results
    };
}

const connectedHomePage = connect(mapStateToProps)(SurveyResultsAnswerPage);
export { connectedHomePage as SurveyResultsAnswerPage };

SurveyResultsAnswerPage.defaultProps = {
    results: {},
    show: () => {},
    handleConfirm: () => {},
    handleCancel: () => {},
    dispatch: () => {}
}

SurveyResultsAnswerPage.propTypes = {
    handleConfirm: PropTypes.func.isRequired,
    handleCancel: PropTypes.func.isRequired,
    dispatch: PropTypes.func.isRequired,
    results: PropTypes.object.isRequired,
    show: PropTypes.func.isRequired,
}


