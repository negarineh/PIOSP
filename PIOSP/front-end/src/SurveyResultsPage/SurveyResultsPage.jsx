import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import '../components/stylesheets/menuStyle.css';
import 'react-tabs/style/react-tabs.css';

// Import React Table
import ReactTable from "react-table";
import "react-table/react-table.css";

import matchSorter from 'match-sorter';
import _ from "lodash";

import { Segment, Button, Dropdown} from 'semantic-ui-react';

import {CSVLink} from 'react-csv';

  const tagOptionsMenu = [
    {
      text: 'Survey Results Based on Category',
      value: 'Survey Results Based on Category',
      icon: {name:'table', color:'black'}
    },
    {
      text: 'Survey Results',
      value: 'Survey Results',
      icon: {name:'table', color:'black'},
    },
  ];

// all answers for submit page in a table by query options like name, id, date,.etc
class SurveyResultsPage extends React.Component {

    constructor(props) {
        super(props);
        this.state = {
          expanded: false,
          photo:'',
          answers:[],
          activeItem: 'home',
          option:'',
        }; 
      }

    componentDidMount() {
        
        fetch('/surveyResultsUpdate') // ('/reportAnswers')
        .then(data => data.json())
        .then((data) => { this.setState({ answers: data });
      }).catch(err => console.log(err));
        
    }

    handleExpandChange = (expanded) => {
        this.setState({expanded: expanded});
      };
    

    population(c) { return c.population; }

    handleItemClick = (e, { name }) => {
        this.setState({ activeItem: name });
    };

    handleChange = (e, { name, value }) => 
        {
            this.setState({ [name]: value , option: value});
            if (value === 'Profile')
                alert("Profile");
            if (value === 'SignOut')
                alert('SignOut');
        };
  
    render() {
        const { answers, option } = this.state;

        return (
            <div >
                <Segment style={{height: '638.5px'}}>
                    <h3>All surveys:</h3>
                    {answers &&
                    <div>
                        <label style={{textAlign:'left'}}>Select Table</label>
                        <Dropdown
                            selection
                            name='default'
                            options={tagOptionsMenu}
                            placeholder='Select Table'
                            onChange={this.handleChange.bind(this)}
                            onClick={this.handleItemClick.bind(this)}
                            fluid
                        />
                    {option ==='Survey Results Based on Category'?
                        <div className="wrapper" > 
                                <ReactTable style={{backgroundColor:"beige",maxHeight:'450px'}}
                                    data={answers}
                                    filterable
                                    defaultFilterMethod={(filter, row) =>
                                    String(row[filter.id]) === filter.value}
                                    columns={[
                                        {
                                        columns: [
                                            {
                                            Header: "ID",
                                            id: "Id",
                                            accessor: d => d.Id,
                                            filterMethod: (filter, rows) =>
                                            matchSorter(rows, filter.value, { keys: ["Id"] }),
                                            filterAll: true,
                                            },
                                            {
                                            Header: "Photo",
                                            id: "photo",
                                            accessor: d => d.photo,
                                            filterMethod: (filter, rows) =>
                                            matchSorter(rows, filter.value, { keys: ["photo"] }),
                                            filterAll: true,
                                            },
                                        ]
                                        },
                                            {
                                        columns: [
                                            {
                                            Header: "Answer",
                                            id: "answer",
                                            accessor: d => d.answer,
                                            filterMethod: (filter, rows) =>
                                            matchSorter(rows, filter.value, { keys: ["answer"] }),
                                            filterAll: true,
                                            },
                                            {
                                            Header: "Description",
                                            id: "description",
                                            accessor: d => d.description,
                                            filterMethod: (filter, rows) =>
                                            matchSorter(rows, filter.value, { keys: ["description"] }),
                                            filterAll: true,
                                            Footer: (
                                                <span>
                                                <strong>Longest:</strong>{" "}
                                                {_.reduce(
                                                    _.map(
                                                    _.groupBy(answers, d => d.description),
                                                    (d, key) => key
                                                    ),
                                                    (a, b) => (a.length > b.length ? a : b)
                                                )}
                                                </span>
                                                    )   
                                            }
                                        ]
                                        }
                                    ]}
                                    pivotBy={[ "answer"]}
                                    defaultPageSize={10}
                                    className="-striped -highlight"
                                    />
                        
                            </div>:
                        null }
                    </div>}
                            {option ==='Survey Results'?
                            <div className="wrapper" > {/*style={{paddingLeft:"40px", paddingRight:"40px"}} */}
                            <ReactTable style={{backgroundColor:"beige", maxHeight:'450px'}}
                            data={answers}
                            filterable
                            defaultFilterMethod={(filter, row) =>
                            String(row[filter.id]) === filter.value}
                            columns={[
                                {
                                columns: [
                                    {
                                    Header: "ID",
                                    id: "Id",
                                    accessor: d => d.Id,
                                    filterMethod: (filter, rows) =>
                                    matchSorter(rows, filter.value, { keys: ["Id"] }),
                                    filterAll: true
                                    },
                                    {
                                    Header: "Photo",
                                    id: "photo",
                                    accessor: d => d.photo,
                                    filterMethod: (filter, rows) =>
                                    matchSorter(rows, filter.value, { keys: ["photo"] }),
                                    filterAll: true
                                    }
                                ]
                                },
                                    {
                                columns: [
                                    {
                                    Header: "Answer",
                                    id: "answer",
                                    accessor: d => d.answer,
                                    filterMethod: (filter, rows) =>
                                    matchSorter(rows, filter.value, { keys: ["answer"] }),
                                    filterAll: true
                                    },
                                    {
                                    Header: "Description",
                                    id: "description",
                                    accessor: d => d.description,
                                    filterMethod: (filter, rows) =>
                                    matchSorter(rows, filter.value, { keys: ["description"] }),
                                    filterAll: true
                                    }
                                ]
                                }
                            ]}
                            defaultPageSize={10}
                            className="-striped -highlight"
                            showPaginationTop
                            showPaginationBottom
                            />
                            {'Tip: Hold shift when sorting to multi-sort!'}
                            </div>:
                            null }
                        <Button style={{margin:'20px'}}>
                            <CSVLink data={answers} >Download Survey Results CSV file</CSVLink>
                        </Button>
                </Segment>
            </div>
        );
    }
}

function mapStateToProps(state) {
    const { users, authentication } = state;
    const { user } = authentication;
    return {
        user,
        users
    };
}

const connectedHomePage = connect(mapStateToProps)(SurveyResultsPage);
export { connectedHomePage as SurveyResultsPage };

SurveyResultsPage.defaultProps = {
    handleItemClick: () => {},
    handleChange: () => {}
}

SurveyResultsPage.propTypes = {
    handleItemClick: PropTypes.func.isRequired,
    handleChange: PropTypes.func.isRequired,
}