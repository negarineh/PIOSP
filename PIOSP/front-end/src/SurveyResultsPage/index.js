export * from './SurveyResultsPage';
export * from './SurveyResultsAnswerPage';
export * from './NumberOfAnswersCategoryChart';
export * from './NumberOfCorrectAnswersCategory';
export * from './NumberOfCorrectAnswers';