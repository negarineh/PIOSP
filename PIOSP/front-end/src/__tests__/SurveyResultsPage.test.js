import React from "react";
import Enzyme, { shallow } from "enzyme";
import assert from "assert";
import {SurveyResultsPage} from "../SurveyResultsPage";
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

import { store } from '../_helpers';
import { Provider } from 'react-redux';
 
Enzyme.configure({ adapter: new Adapter() });

// unit tests for the UserDeletePrompt component
describe('Survey Results Page component', () => {
  describe('render()', () => {
    it('should render the component', () => {
      const props = {handleChange: () => {}, handleItemClick: () => {}};
      const wrapper = shallow(<Provider store={store}><SurveyResultsPage {...props}/></Provider>);
      const tree = toJson(wrapper);
      expect(tree).toMatchSnapshot();
      assert.equal(wrapper.length, 1);
    });
  });
});
