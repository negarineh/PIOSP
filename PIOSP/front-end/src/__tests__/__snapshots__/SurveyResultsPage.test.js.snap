// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Survey Results Page component render() should render the component 1`] = `
<Connect(SurveyResultsPage)
  handleChange={[Function]}
  handleItemClick={[Function]}
/>
`;
